### ISSUE TITLE
Package dependence

### ISSUE THREAD

**Issue by JianghuiDu** [JuliaLang/PackageCompiler.jl#223]
I'm compiling a few packages, and when the processes are about to finish `PackageCompiler` almost always reports dependence errors such as (this is when compiling `DataFrames`)
```
ERROR: LoadError: LoadError: ArgumentError: Package DataValues [e7dc6d0d-1eca-5fa6-8ad6-5aecde8b7ea5] is required but does not seem to be installed:
 - Run `Pkg.instantiate()` to install all recorded dependencies.
```
These packages are not needed when using the packages compiled and I'm not sure why they are dependent (some appear necessary for testing). And  run `Pkg.instantiate()` doesn't do anything. Is it possible to check such errors at the beginning so we can install the dependent packages before compiling and not wasting time?

**Comment by arnavs** [comment]
Has there been any solution to this issue? (the fact that PackageCompiler is failing to grab test dependencies.)

**Comment by makslevental** [comment]
yes i hit this same thing over and over and i don't understand why these dependencies can't just be installed

**Comment by arnavs** [comment]
@KristofferC mentioned something about test sub-projects at JuliaCon. This could be a long-term solution to this issue (test is a sub-project which can be instantiated.)

**Comment by makslevental** [comment]
ok but how do i solve this problem now? how do i get all the way through a compile without having to restart every time there's a missing dependency?

**Comment by kdheepak** [comment]
Did you try using the `-autodeps` flag when you use PackageCompiler? If so, can you post a minimal working example here of this issue?

**Comment by makslevental** [comment]
@kdheepak 

i'm running 

`compile_package("Plots", "PyPlot", "DataFrames", force = false)`

how do i pass autodeps here?

**Comment by kdheepak** [comment]
It appears that `autodeps` is always set to true when running `build_sysimg` which is what is called in `compile_package`. So that's not the solution.
